src/app/shared/pagination/pagination-helper.ts

```diff
@@ -0,0 +1,34 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { PaginatedResult } from './pagination';
+import { map } from 'rxjs';
+
+export function getPaginatedResult<T>(url: string, params: HttpParams, httpClient: HttpClient) {
+  const paginatedResult: PaginatedResult<T> = new PaginatedResult<T>();
+
+  return httpClient.get<any>(url, { observe: 'response', params }).pipe(
+    map((response) => {
+      if (response.body) {
+        paginatedResult.result = response.body.page;
+        paginatedResult.pagination = {
+          currentPage: response.body.currentPage,
+          itemsPerPage: response.body.page.length,
+          totalItems: response.body.totalElements,
+          totalPages: response.body.totalPages
+        };
+      }
+      return paginatedResult;
+    })
+  );
+}
+
+export function getPaginationParams<T>(paramsObject: T): HttpParams {
```

Comment:
You can create instance of HttpParams from object by passing it into constructor like this
HttpParams({fromObject: {...object}})